### PR TITLE
Have PyTorch files download from model stubs in YOLO

### PR DESCRIPTION
**Pull request by mgoin**
Feature request based on https://github.com/neuralmagic/deepsparse/issues/227

### CODE REVIEW

File: examples/ultralytics-yolo/deepsparse_utils.py [hunk 1]
```diff
@@ -549,6 +550,30 @@ def download_model_if_stub(path: str) -> str:
...
+    """
+    Utility method to download PyTorch model if path is a SparseZoo stub
+
+    :param path: file path to YOLO .pt model or SparseZoo stub of the model
+        to be loaded
+    :return: filepath to the .pt model
+    """
+    if path.startswith("zoo"):
+        model = Zoo.load_model_from_stub(path)
+        downloded_path = model.onnx_file.downloaded_path()
```
Comment (bfineran) [inline examples/ultralytics-yolo/deepsparse_utils.py:563]: nit: typo "downloaded"

Comment (bfineran) [inline examples/ultralytics-yolo/deepsparse_utils.py:563]: what is the purpose of downloading the onnx file here?

File: examples/ultralytics-yolo/deepsparse_utils.py [hunk 2]
```diff
@@ -549,6 +550,30 @@ def download_model_if_stub(path: str) -> str:
...
+        downloaded_pt_path = None
+        for file in model.framework_files:
+            if file.file_type_framework and file.display_name == "model.pt":
+                downloaded_pt_path = file
+        if downloaded_pt_path is None:
+            raise ValueError(
+                f"model with stub {path} has no 'model.pt' associated for PyTorch"
+            )
+        print(f"model with stub {path} downloaded to {downloaded_pt_path}")
+        return downloded_path
```
Comment (bfineran) [inline examples/ultralytics-yolo/deepsparse_utils.py:573]: should return the `downloaded_pt_path` here

File: examples/ultralytics-yolo/deepsparse_utils.py [hunk 3]
```diff
@@ -549,6 +550,30 @@ def download_model_if_stub(path: str) -> str:
...
+        to be loaded
+    :return: filepath to the .pt model
+    """
+    if path.startswith("zoo"):
+        model = Zoo.load_model_from_stub(path)
+        downloded_path = model.onnx_file.downloaded_path()
+        downloaded_pt_path = None
+        for file in model.framework_files:
+            if file.file_type_framework and file.display_name == "model.pt":
+                downloaded_pt_path = file
```
Comment (bfineran) [inline examples/ultralytics-yolo/deepsparse_utils.py:567]: might have missed something, but I believe we still need to download the file here

**Review by bfineran — CHANGES_REQUESTED** [review]

File: examples/ultralytics-yolo/deepsparse_utils.py (hunk 1 above)
Comment (mgoin) [reply examples/ultralytics-yolo/deepsparse_utils.py:563]: hah good catch, this was a mistake left in from copying

File: examples/ultralytics-yolo/deepsparse_utils.py (hunk 3 above)
Comment (mgoin) [reply examples/ultralytics-yolo/deepsparse_utils.py:567]: i thought accessing `framework_files` would auto-download but i will make sure here

**Review by markurtz — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**